### DISCUSSION TITLE
setup reverse proxy for CVAT server

### DISCUSSION THREAD

**Discussion by zvant** [cvat-ai/cvat#4042] [Q&A]
I am trying to setup a reverse proxy for CVAT server. In a network, I have two servers, say `A.net` and `B.net`. Only `A.net` can be accessed from outside, and it has Apache2 installed. On `B.net`, I have CVAT server configured and running at port `8080`. Because `B.net` cannot be seen from outside, I setup a reverse proxy on `A.net` with the Apache2 configuration
`ProxyPass "/cvat"  "http://B.net:8080"`
`ProxyPassReverse "/cvat"  "http://B.net:8080"`

I have tried to visit `http://B.net:8080` from `A.net`, and everything looks fine. However, when I tried to visited `http://A.net/cvat` from outside, it can only show the title "Computer Vision Annotation Tool", but no page content. The browser concole shows a 404 error:
`GET http://A.net/cvat-ui.580642541947bec0314e.min.js net::ERR_ABORTED 404 (Not Found)`
Can someone please tell me what I did wrong in the configurations? Thanks.

**Comment by bsekachev** [comment]
I see it tries to get JS bundle from:

`` http://A.net/cvat-ui.580642541947bec0314e.min.js``, not from `` http://A.net/cvat/cvat-ui.580642541947bec0314e.min.js``. Can it be a reason?

**Reply by bsekachev** [reply]
If you take a look into generated index.html (cvat-ui/dist/index.html)
In my case it has for example:

```
<script type="text/javascript" src="/cvat-ui.2120f64655f5abf28675.min.js"></script></body>
```

JS bundle URL is absolute and it ignores URL path ``cvat/``. Probably webpack can be configured to fix it. 

But all what I said are only my guesses.
